compiler/gentype/GenTypeCommon.ml

```diff
@@ -180,7 +180,7 @@ let ident ?(builtin = true) ?(type_args = []) name =
   Ident {builtin; name; type_args}
 
 let sanitize_type_name name =
-  name
+  name |> Ext_ident.unwrap_uppercase_exotic
```

Comment:
this fixes `type \"UppercaseName"` being emitted literally, breaking the typescript syntax